internal/provider/resource_vdb.go

```diff
@@ -1145,6 +1174,16 @@ func helper_provision_by_bookmark(ctx context.Context, d *schema.ResourceData, m
 	if v, has_v := d.GetOk("tags"); has_v {
 		provisionVDBFromBookmarkParameters.SetPostStop(toHookArray(v))
 	}
+	if v, has_v := d.GetOk("appdata_source_params"); has_v {
+		appdata_source_params := make(map[string]interface{})
+		json.Unmarshal([]byte(v.(string)), &appdata_source_params)
+		provisionVDBFromBookmarkParameters.SetAppdataSourceParams(appdata_source_params)
+	}
+	if v, has_v := d.GetOk("appdata_config_params"); has_v {
+		appdata_config_params := make(map[string]interface{})
+		json.Unmarshal([]byte(v.(string)), &appdata_config_params)
+		provisionVDBFromBookmarkParameters.SetAppdataConfigParams(appdata_config_params)
+	}
```

Comment:
Note that while this is technically correct and supports all appdata based platforms, this is obviously not very user friendly. A more user friendly approach, if the goal is to optimize for postgresql / hana, would be to have postgresql / hana specific properties defined in the schema, and convert them here to `appdata_config_params` / `appdata_source_params`.

Reply:
So we thought about it and the thing is those parameters that are defined within the ```appdata_source_params``` can be referenced by different key for the same parameter. For eg: In Postgres, "configSettingsStg" is the same parameter for "configServices". So there will be very specific parameters for different databases which ultimately means the same thing. That is one problem. Another problem would be if any changes to even a small trivial parameter name, we would have to end up doing the same change in the provider code as well. So hence we made it more generic so that we can cater to a more wider array of Appdata plugins. 
However, having said that, do you see a challenge in this approach? or maybe something we might have overlooked?

Reply:
resolving it

Reply:
I think your analysis is right, basically it comes down to choosing if we want to make the extra effort to make appdata feel native (more work for Delphix developers, end result is easy to use for end users) or if we're okay to have appdata be hard to use (no ways customers can't use it as-is, nobody knows the parameters to pass in, will have to copy paste examples from Delphix team) so that it is easy to maintain.

In DCT core, we initially started with the generic approach, which remains present, but had sufficient pressure from the field to make postgresql feel native that we've now also added postgresql to the natively supported platforms (but not actively working on doing this for other appdata platforms until there's enough traction).